state/application_test.go

```diff
@@ -1583,9 +1583,27 @@ func (s *ApplicationSuite) TestSettingsRefRemoveRace(c *gc.C) {
 	assertSettingsRef(c, s.State, appName, newCh, 1)
 }
 
+// assertNoRemoteApplications checks that we have cleaned up all remote
+// application for a given application, specified by its name.
+func assertNoRemoteApplications(c *gc.C, st *state.State, appName string) {
+	// can't access the application's relations via st.Application
```

Comment:
It's maybe sufficient just to try and get the remote application by name and expect a NotFound error?